libvast/vast/fbs/layout.fbs

```diff
@@ -0,0 +1,11 @@
+include "type.fbs";
+
+namespace vast.fbs;
+
+table Layout {
+  name: string;
+  record: RecordType;
+  attributes: [Attribute];
+}
```

Comment:
@tobim @dominiklohmann What do you think of a separate type for layouts, like this?

Reply:
Do you think it makes sense to change our layout behavior at the same time as switching to flatbuffers? It might make more sense to introduce this change afterwards.

Reply:
I think it can happen independently. But a new type `vast::layout` would be useful only once we've made the change with type names and alias types.

Reply:
As discussed, we can keep this.